vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include <stdio.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string.h>
+
+using namespace std;
+typedef float weight;
+
+class array_parameters
+{
+private:
+	weight* _begin;
+
+public:
+	array_parameters(weight* begin)
+		: _begin(begin)
+	{ }
+
+	array_parameters(const array_parameters& other)
+		: _begin(other._begin)
+	{ }
+	
+	weight& operator[](size_t i) const { return _begin[i]; }
+
+	inline weight*& begin() { return _begin; }
+
+	template<class T> array_parameters operator+(T index) { return array_parameters(_begin + index); }
+
+	array_parameters& operator++()
+	{
+		_begin++;
+		return *this;
```

Comment:
This operator seems dangerous to me.  You either do a copy/increment/discard or craziness happens when _begin increments.   I think we want an iterator which is best done using stride and with a different type.